code/modules/reagents/chemistry/machinery/pandemic.dm

```diff
@@ -225,25 +225,6 @@
 		ui.open()
 
 /obj/machinery/computer/pandemic/ui_data(mob/user)
-	var/datum/reagent/blood/Blood = null
-	if(beaker)
-		var/datum/reagents/R = beaker.reagents
-		for(var/datum/reagent/blood/B in R.reagent_list)
-			if(B)
-				Blood = B
-				break
-
-	var/list/data = list(
-		"synthesisCooldown" = wait ? TRUE : FALSE,
-		"beakerLoaded" = beaker ? TRUE : FALSE,
-		"beakerContainsBlood" = Blood ? TRUE : FALSE,
-		"beakerContainsVirus" = length(Blood?.data["viruses"]) != 0,
-		"selectedStrainIndex" = selected_strain_index,
-	)
-
-	return data
-
-/obj/machinery/computer/pandemic/ui_static_data(mob/user)
```

Comment:
I have to think about this one. For this fix, I think having it in `ui_data` is fine. Having it in `ui_data` improves updating the name when someone names a virus. We are now also using the length of strains to check if a beaker contains a virus. I think it would be more of a concern if the Pandemic was being used heavily or if it was sending a lot of information to begin with. I understand where you are coming from though.

Reply:
I like to keep as much stuff in static data as possible to keep performance costs down. If you're concerned about the speed of updating the UI, you could call `update_static_data(user)`.

Reply:
Discussed this a bit more in depth on Discord. I am going to revert to the original fix and handle the length check for Strain on the JS side, moving the other data back to `ui_static_data`.

Reply:
Updated in latest commit.